src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/dto/chat/PyrisChatStatusUpdateDTO.java

```diff
@@ -2,11 +2,13 @@
 
 import java.util.List;
 
+import jakarta.annotation.Nullable;
+
 import com.fasterxml.jackson.annotation.JsonInclude;
 
 import de.tum.cit.aet.artemis.core.domain.LLMRequest;
 import de.tum.cit.aet.artemis.iris.service.pyris.dto.status.PyrisStageDTO;
 
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record PyrisChatStatusUpdateDTO(String result, List<PyrisStageDTO> stages, List<String> suggestions, List<LLMRequest> tokens) {
+public record PyrisChatStatusUpdateDTO(@Nullable String result, List<PyrisStageDTO> stages, List<String> suggestions, @Nullable List<LLMRequest> tokens) {
```

Comment:
For `result` and `tokens` there are null checks in the code, so I set them to nullable - for the other variables I am not sure if null checks might just be missing of they should really be notnull